src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -325,6 +326,12 @@ public void SniffIncomingMessage(Message message)
                         invokeExceptionLogger.Warn(ErrorCode.GrainInvokeException,
                             "Exception during Grain method call of message: " + message, exc1);
                     }
+
+                    if (exc1 is InconsistentStateException && target is Grain)
+                    {
+                        this.DeactivateOnIdle(((Grain)target).Data.ActivationId);
```

Comment:
should we log something, given that it's a new (and potentially surprising) behavior?